firestore-palm-summarize-text/functions/src/generator.ts

```diff
@@ -54,33 +59,101 @@ export class TextGenerator {
     this.candidateCount = options.candidateCount;
     this.instruction = options.instruction;
     if (options.model) this.model = options.model;
-    logs.usingADC();
-    const auth = new GoogleAuth({
-      scopes: [
-        'https://www.googleapis.com/auth/userinfo.email',
-        'https://www.googleapis.com/auth/generative-language',
-      ],
-    });
-    this.client = new TextServiceClient({
-      auth,
-    });
+
+    this.endpoint = `projects/${config.projectId}/locations/${config.location}/publishers/google/models/${this.model}`;
+
+    if (config.useVertex) {
+      const clientOptions = {
+        apiEndpoint: `${config.location}-prediction-aiplatform.googleapis.com`,
+      };
+
+      this.vertexClient = new v1.PredictionServiceClient(clientOptions);
+    } else {
+
+      logs.usingADC();
+
+      const auth = new GoogleAuth({
+        scopes: [
+          'https://www.googleapis.com/auth/userinfo.email',
+          'https://www.googleapis.com/auth/generative-language',
+        ],
+      });
+      this.generativeClient = new TextServiceClient({
+        auth,
+      });
+    }
   }
 
   async generate(
-    prompt: string,
+    promptText: string,
```

Comment:
Do we have any tests included for this logic, maybe we can reuse from another projects implemetnation.